kernel/src/main/scala/cats/Eq.scala

```diff
@@ -0,0 +1,106 @@
+package cats
+
+import scala.{specialized => sp}
+
+import scala.math.Equiv
+
+/**
+ * A type class used to determine equality between 2 instances of the same
+ * type. Any 2 instances `x` and `y` are equal if `eqv(x, y)` is `true`.
+ * Moreover, `eqv` should form an equivalence relation.
+ */
+trait Eq[@sp A] extends Any with Serializable { self =>
```

Comment:
is there any value in making Eq contravariant? It could be (with proper modifications).

It might help get `Eq[Some[Int]]` inferred because we should use `Eq[Option[Int]]` in that case. But I can't think of others.